src/wanderer/yellowray.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  canAdventure,
+  getLocationMonsters,
+  itemDropsArray,
+  Location,
+  print,
+  toMonster,
+} from "kolmafia";
+import { sumNumbers } from "libram";
+import { garboValue } from "../session";
+import { canWander, DraggableFight, UnlockableZones, WandererTarget } from "./lib";
+
+function averageYrValue(location: Location, debug = false) {
+  const monsters = Object.keys(getLocationMonsters(location))
+    .map((m) => toMonster(m))
+    .filter((m) => !["LUCKY", "ULTRARARE", "BOSS"].some((s) => m.attributes.includes(s)));
+
+  if (monsters.length === 0) {
+    return 0;
+  } else {
+    return (
+      sumNumbers(
+        monsters.map((m) => {
+          const items = itemDropsArray(m)
+            .filter((drop) => ["", "n"].includes(drop.type))
+            .map((drop) => garboValue(drop.drop));
+
+          if (debug) print(`${m}: ${items.join(",")}`);
+
+          return sumNumbers(items);
+        })
+      ) / monsters.length
+    );
+  }
+}
+
+function yrValues(): Map<Location, number> {
+  const values = new Map<Location, number>();
+  for (const location of Location.all()) {
```

Comment:
should we filter down `Location.all()` at all?

I have a change in the pipeline that will let us use historical values for this, which would help with the performance here

Reply:
I am now more confident that we should at least slightly filter `Location.all()` by removing underwater zones, zones from the SkipList, and zones with a combat rate of less than... I guess `70 - getModifier("Combat Rate")`? 70 coming from 100 - the soft cap of 30. 

`averageYrValue` currently doesn't use historical values, so we really want to pare down the number of zones we search to only be ones we can plausibly select. Even with historical values it is probably worth doing, because we'll still trigger mallsearches weeklyish.